### ISSUE TITLE
Help getting spark-connect-rs running locally

### ISSUE THREAD

**Issue by MrPowers** [sjrusso8/spark-connect-rs#14]
Commands I ran (I am a total n00b):

* install rust via `rustup`
* Created a new crate: `cargo new spark-connect-gist`
* edited `Cargo.toml` to include spark connect rs and `tokio` dependencies:

```
[dependencies]
spark-connect-rs = "0.0.1-beta.3"
tokio = { version = "1", features = ["full"] }
```

* Install cmake: `arch -arm64 brew install cmake`
* Install protobuf: `brew install protobuf`
* Copy the example gist in `main.rs`: https://gist.github.com/sjrusso8/2b4e43af462367a15f91db5a33627449
* Run `cargo run`

Here is the error message:

```
   Compiling spark-connect-gist v0.1.0 (/Users/matthew.powers/Documents/code/my_apps/spark-connect-gist)
    Finished dev [unoptimized + debuginfo] target(s) in 1.00s
     Running `target/debug/spark-connect-gist`
Error: tonic::transport::Error(Transport, hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 61, kind: ConnectionRefused, message: "Connection refused" })))
```

**Comment by MrPowers** [comment]
Whoops, this is actually working, I forgot to start the Spark Connect session with the command: `./sbin/start-connect-server.sh --packages org.apache.spark:spark-connect_2.12:3.5.1`

Are you OK if I add these setup instructions to the README?

**Comment by sjrusso8** [comment]
So I just ran through the steps on a fresh ubuntu image, and here is what I had to do. 

- Install rust via `rustup`
- Install `build-essential` & `protoc` with `apt install -y build-essential protobuf-compiler`
- Install `cmake`
- Run `cargo new spark-connect-starter && cd spark-connect-starter` to create a new project and go to the directory
- Add dependencies to your `Cargo.toml` file

```
spark-connect-rs = "0.0.1-beta.3"
tokio = "1.37.0"
```
- Copy example into `main.rs` and run `cargo run`

Ensure spark connect is up and running.

-------

I'll update the top level readme with the example from the gist. The current one leverages a file path that might not exist for all users depending on how they are running their spark cluster.
